### PR TITLE
Add FindByPid VersionController

### PR DESCRIPTION
**Pull request by BPScott**
Visting /programmes/:versionPid now redirects you to the episode that the
version belongs to

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Controller/FindByPid/VersionController.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * Redirect to the Episode that the version belongs to
+ */
+class VersionController extends BaseController
+{
+    public function __invoke(Version $version)
+    {
+        $episodePid = (string) $version->getProgrammeItem()->getPid();
+        $response = $this->redirectToRoute('find_by_pid', [ 'pid' => $episodePid], 303);
+        $response->setPublic()->setMaxAge(120);
```
Comment (votemike) [inline src/Controller/FindByPid/VersionController.php]: Why only 2 minutes?

File: src/Controller/FindByPid/VersionController.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+use BBC\ProgrammesPagesService\Domain\Entity\Version;
+
+/**
+ * Version Page
+ *
+ * Redirect to the Episode that the version belongs to
+ */
+class VersionController extends BaseController
+{
+    public function __invoke(Version $version)
```
Comment (votemike) [inline src/Controller/FindByPid/VersionController.php]: Return typehint?

File: tests/EventSubcriber/FindByPidRouterSubscriberTest.php [hunk 3]
```diff
@@ -54,8 +57,10 @@ public function coreEntityDataProvider()
             [$series, null, null, 'App\Controller\FindByPid\DefaultController'],
             [$this->createMock(Episode::class), null, null, 'App\Controller\FindByPid\DefaultController'],
             [$this->createMock(Clip::class), null, null, 'App\Controller\FindByPid\DefaultController'],
+            // TODO add checks for Groups
```
Comment (votemike) [inline tests/EventSubcriber/FindByPidRouterSubscriberTest.php]: Groups exist don't they? Why is this a TODO?

Comment (BPScott) [inline tests/EventSubcriber/FindByPidRouterSubscriberTest.php]: Shall be done once I have a coreEntitiesService that I can use to make a query, and get data from :)

File: src/Controller/FindByPid/VersionController.php (hunk 2 above)
Comment (BPScott) [inline src/Controller/FindByPid/VersionController.php]: We haven't bothered setting them on any of our other Controllers.

File: src/Controller/FindByPid/VersionController.php (hunk 1 above)
Comment (BPScott) [inline src/Controller/FindByPid/VersionController.php]: Standard cache lifetime of our pages. This is the duration of time that a page is cached in Varnish too.

Admittedly it's heinously that a Version shall change the programme it is attached to, but it is useful for that to be reflected reasonably quickly.

File: src/Controller/FindByPid/VersionController.php (hunk 2 above)
Comment (votemike) [inline src/Controller/FindByPid/VersionController.php]: Great time to start 👍

Comment (BPScott) [inline src/Controller/FindByPid/VersionController.php]: I don't think it's really worth it tbh, it's an extra use statement and a typehint that I don't think adds any meaningful value

**Pull request merged**